third_party/terraform/utils/batcher.go

```diff
@@ -0,0 +1,197 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"sync"
+	"time"
+)
+
+const defaultBatchSendIntervalSec = 10
+
+// RequestBatcher a global batcher object.
+type RequestBatcher struct {
+	sync.Mutex
+
+	*batchingConfig
+	batches map[string]*startedBatch
+}
+
+// BatchRequest represents a single request to a global batcher.
+type BatchRequest struct {
+	// ResourceName determine the resource name to be passed to SendF.
```

Comment:
It's not immediately obvious to me which resource name this should be, whether it's a TF or GCP resource name. This will end up being the parent resource (such as the project below) a lot of the time too, right?

Reply:
Yes - its mainly a way for callers to have to avoid folding it into the body, but its format is specific to a request type, so vague here. I'll try to clarify comment.